### PR TITLE
feat(scan): add scan duration

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

Scan duration is needed for the API usage. This PR adds this property to the `Scan` class.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/scan/scan.py [hunk 1]
```diff
@@ -18,6 +19,7 @@ class Scan:
     _service_checks_completed: dict[str, set[str]]
     _progress: float = 0.0
     _findings: list = []
+    _duration: float = 0.0
```
Comment (jfagoagas) [inline prowler/lib/scan/scan.py]: I think an `int` is more than enough here.

File: prowler/lib/scan/scan.py [hunk 2]
```diff
@@ -159,6 +166,8 @@ def scan(
                     logger.error(
                         f"{check_name} - {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                     )
+            # Update the scan duration when all checks are completed
+            self._duration = (datetime.datetime.now() - start_time).total_seconds()
```
Comment (jfagoagas) [inline prowler/lib/scan/scan.py]: Can we remove the microsencond precision, more than enough with seconds.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/lib/scan/scan.py (hunk 1 above)
Comment (pedrooot) [reply prowler/lib/scan/scan.py]: Done

File: prowler/lib/scan/scan.py (hunk 2 above)
Comment (pedrooot) [reply prowler/lib/scan/scan.py]: Done

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**